crypto/secp256k1/encoding.go

```diff
@@ -0,0 +1,122 @@
+package secp256k1
+
+import (
+	"bytes"
+	"fmt"
+
+	amino "github.com/tendermint/go-amino"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+const (
+	PrivKeyAminoName = "tendermint/PrivKeySecp256k1"
+	PubKeyAminoName  = "tendermint/PubKeySecp256k1"
+)
+
+var cdc = amino.NewCodec()
+
+func init() {
+	cdc.RegisterInterface((*crypto.PubKey)(nil), nil)
+	cdc.RegisterConcrete(PubKey{},
+		PubKeyAminoName, nil)
+
+	cdc.RegisterInterface((*crypto.PrivKey)(nil), nil)
+	cdc.RegisterConcrete(PrivKey{},
+		PrivKeyAminoName, nil)
+}
+
+var (
+	prefixPrivKeySecp256k1      = []byte{0xE1, 0xB0, 0xF7, 0x9B}
+	lengthPrivKeySecp256k1 byte = 0x20
+
+	prefixPubKeySecp256k1      = []byte{0xEB, 0x5A, 0xE9, 0x87}
+	lengthPubKeySecp256k1 byte = 0x21
+)
+
+// MarshalBinary attempts to marshal a PrivKeySecp256k1 type that is backwards
+// compatible with Amino.
```

Comment:
Oh I see, this is used to store the private keys locally, right?

Reply:
these are here for now, as I go through the codebase to find where we actually depend on `MarshalLengthPrefixed` (currently can't find anything), but if there isn't a need for them then they will be removed. These can also be helper functions if any apps are relying on `LengthPrefixed` bytes